### PR TITLE
Remove querying all builds to speed up performance

### PR DESCRIPTION
**Pull request by freeqaz**
Currently this adds about 10 seconds in production and slams the DB. This fixes that.

### CODE REVIEW

**Review by ajvpot — APPROVED** [review]

File: lunatrace/bsl/frontend/src/components/navbar/NavbarBreadcrumbs.tsx [hunk 1]
```diff
@@ -63,25 +63,34 @@ const ProjectBreadCrumb: BreadcrumbComponentType = (crumbProps: BreadcrumbCompon
...
+    console.error('Error loading build number', new Error('Error loading build number'));
+    return <span>Error loading build number</span>;
+  }
+
+  const buildsByPrimaryKey = data.builds_by_pk;
+
+  if (!buildsByPrimaryKey) {
+    console.error(
+      'Error, could not find build by primary key',
+      new Error('Error, could not find build by primary key')
```
[the remark this answers is not in the document]
Comment (factoidforrest) [reply lunatrace/bsl/frontend/src/components/navbar/NavbarBreadcrumbs.tsx:88]: Does it look okay when we do a spinner in a breadcrumb? I usually run pages with the network limiter at a ping of 300 or so to make sure it looks alright for prod

**Review by factoidforrest — APPROVED** [review]

File: lunatrace/bsl/frontend/src/api/graphql/getSidebarInfo.graphql [hunk 2]
```diff
@@ -19,10 +15,6 @@ query GetSidebarInfo($kratos_id: uuid) {
       github_repository {
         id
       }
-      builds {
```
[the remark this answers is not in the document]
Comment (freeqaz) [reply lunatrace/bsl/frontend/src/api/graphql/getSidebarInfo.graphql:22]: Yeah it is very slow in prod which is why we were doing this at all (every page has 10+ seconds of latency when I browse it)

File: lunatrace/bsl/frontend/src/components/navbar/NavbarBreadcrumbs.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (freeqaz) [reply lunatrace/bsl/frontend/src/components/navbar/NavbarBreadcrumbs.tsx:88]: Screenshot_2022-09-05_19-04-26

I think it looks fine -- thanks for the ping info

File: lunatrace/bsl/frontend/src/api/index.ts [hunk 3]
```diff
@@ -38,6 +38,9 @@ const appApi = generatedApi.enhanceEndpoints({
     GetBuildDetails: {
       providesTags: ['Builds'],
     },
+    GetBuildNumber: {
+      providesTags: ['BuildNumber'],
```
[the remark this answers is not in the document]
Comment (freeqaz) [reply lunatrace/bsl/frontend/src/api/index.ts:42]: Yeah I just nuked this. I didn't quite understand what it was lol

**Pull request merged**